lib/ast/class_def.ts

```diff
@@ -429,21 +429,23 @@ export class EntityDef extends Node {
      */
     constructor(location : SourceRange|null,
                 name : string,
-                _extends : string|null,
+                _extends : string[]|string|null,
                 annotations : AnnotationSpec) {
         super(location);
         this.name = name;
 
-        this.extends = _extends;
+        _extends = typeof _extends === 'string' ? [_extends] : _extends;
+        this.extends = _extends || [];
 
         this.nl_annotations = annotations.nl || {};
         this.impl_annotations = annotations.impl || {};
     }
 
     toSource() : TokenStream {
-        if (this.extends) {
+        if (this.extends.length > 0) {
+            const _extends = this.extends.map((e) => e.includes(':') ? `^^${e}` : e);
             return List.concat('entity', ' ', this.name,
-                'extends', (this.extends.includes(':') ? '^^' + this.extends : this.extends), '\t+',
+                'extends', ..._extends.join(' , ').split(' '), '\t+',
```

Comment:
Uh? You should use List.join here, that's what it is for. Also, you still need the ^^ prefix if the entity name contains :